framework/src/wifi_manager/wifi_manager.c

```diff
@@ -149,336 +341,250 @@ static wifi_utils_result_e stop_dhcp_server(void)
-	wifi_mutex_release(w_mutex);
 
-#ifdef CONFIG_ENABLE_IOTIVITY
-	if (result == WIFI_UTILS_SUCCESS) {
-		__tizenrt_manual_linkset("del");
+void _wifi_utils_disconnect_event(void *arg)
+{
+	WM_LOG_START;
+	_wifimng_msg_s msg = {EVT_STA_DISCONNECTED, NULL};
+	wifi_manager_result_e res = _handle_request(&msg);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		ndbg("handle connect_event fail\n");
 	}
-#endif
-
-	return result;
 }
 
-wifi_manager_result_e wifi_manager_init(wifi_manager_cb_s *wmcb)
-{
-	wifi_utils_result_e result = WIFI_UTILS_SUCCESS;
-	wifi_utils_info info;
 
-	if (wmcb == NULL) {
-			ndbg("Wi-Fi Manager init fail because of no callbacks");
-			return WIFI_MANAGER_FAIL;
+void _wifi_utils_join_event(void *arg)
+{
+	WM_LOG_START;
+	_wifimng_msg_s msg = {EVT_JOIN, NULL};
+	wifi_manager_result_e res = _handle_request(&msg);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		ndbg("handle connect_event fail\n");
 	}
+}
 
-	if (w_info_mutex == NULL) {
-		w_info_mutex = (wifi_mutex *)malloc(sizeof(wifi_mutex));
-		if (w_info_mutex == NULL) {
-			ndbg("w_info_mutex malloc fails.\n");
-			return WIFI_MANAGER_FAIL;
-		}
-		result = wifi_mutex_create(w_info_mutex);
-		if (result != WIFI_UTILS_SUCCESS) {
-			ndbg("w_info_mutex_create fail");
-			goto error_without_mutex_release;
-		}
-	}
 
-	wifi_mutex_acquire(w_info_mutex, WIFI_UTILS_FOREVER);
-	if (w_mutex != NULL) {
-		ndbg("WI-FI is already initialized.\n");
-		wifi_mutex_release(w_info_mutex);
-		return WIFI_MANAGER_INITIALIZED;
+void _wifi_utils_leave_event(void * arg)
+{
+	WM_LOG_START;
+	_wifimng_msg_s msg = {EVT_LEAVE, NULL};
+	wifi_manager_result_e res = _handle_request(&msg);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		ndbg("handle connect_event fail\n");
 	}
+}
+
 
-	w_mutex = (wifi_mutex *)malloc(sizeof(wifi_mutex));
-	if (w_mutex == NULL) {
-		ndbg("w_mutex malloc fails.\n");
-		wifi_mutex_release(w_info_mutex);
-		goto error_with_mutex_destory;
+void _wifi_utils_scan_done(wifi_utils_result_e result, wifi_utils_scan_list_s *slist, void *arg)
+{
+	WM_LOG_START;
+	_wifimng_msg_s msg = {EVT_SCAN_DONE, (void *)slist};
+	wifi_manager_result_e res = _handle_request(&msg);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		ndbg("handle scan done event fail\n");
 	}
+}
 
-	result = wifi_mutex_create(w_mutex);
-	if (result != WIFI_UTILS_SUCCESS) {
-		ndbg("wifi_mutex_create fail");
-		wifi_mutex_release(w_info_mutex);
-		goto error_with_mutex_destory;
+void _wifi_dhcpd_event(void)
+{
+	WM_LOG_START;
+	_wifimng_msg_s msg = {EVT_DHCPD_GET_IP, NULL};
+	wifi_manager_result_e res = _handle_request(&msg);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		ndbg("handle dhcpd_event fail\n");
 	}
 
-	/* wifi_manager mode is switched to wifi_initializing */
-	g_manager_info.pre_mode = WIFI_NONE;
-	g_manager_info.next_mode = STA_MODE;
-	g_manager_info.mode = WIFI_INITIALIZING;
-	wifi_mutex_release(w_info_mutex);
+	return ;
+}
+
+/*
+ * State Functions
+ */
+wifi_manager_result_e _handle_init_state(_wifimng_msg_s *msg)
```

Comment:
The name of _handle_init_stat() is misleading.
I recommend _handle_state_uninitialized() or _state_handler_uninitialized().
In addition, what is the exact meaning of WIFIMNG_INIT state?
Does it mean the wifi-manager is requiring initializing or already initialized?

Reply:
I will fix.